### PR TITLE
[IMP] spreadsheet: update o_spreadsheet to latest version

### PR DESCRIPTION
**Pull request by pro-odoo**
### Contains the following commits:

https://github.com/odoo/o-spreadsheet/commit/96166819 [REF] evaluation: decouple evaluation from core plugin
https://github.com/odoo/o-spreadsheet/commit/a5a67d88 [IMP] find & replace: displays 0 match
https://github.com/odoo/o-spreadsheet/commit/2e2d878f [FIX] overflowing cell with border rendering
https://github.com/odoo/o-spreadsheet/commit/40dfc80c [FIX] interactive merge: remove useless user confirmation
https://github.com/odoo/o-spreadsheet/commit/050e64f3 [IMP] format: support additional date formats

### CODE REVIEW

**Comment by pro-odoo** [comment]
fw-bot ignore

**Comment by pro-odoo** [comment]
@robodoo override=ci/style 
Override because the terms which trigger the warnings (border-*) are used for custom class names or directly in inlined css.

**Comment by pro-odoo** [comment]
robodoo override=ci/style 
Override because the terms which trigger the warnings (border-*) are used for custom class names or directly in inlined css.

File: addons/spreadsheet/static/src/ir_ui_menu/index.js [hunk 1]
```diff
@@ -6,64 +6,141 @@ import spreadsheet from "@spreadsheet/o_spreadsheet/o_spreadsheet_extended";
...
+                    const menu = env.services.menu.getMenu(menuId);
+                    env.services.action.doAction(menu.actionID);
                 },
+                // createCell: (id, content, properties, sheetId, getters) => {
+                //     const { url } = parseMarkdownLink(content);
+                //     const xmlId = parseIrMenuXmlLink(url);
+                //     const menuId = _getIrMenuByXmlId(xmlId).id;
+                //     const menuName = _getIrMenuByXmlId(xmlId).name;
+                //     return new OdooMenuLinkCell(id, content, menuId, menuName, properties);
+                // },
```
Comment (pro-odoo) [inline addons/spreadsheet/static/src/ir_ui_menu/index.js:107]: Commented code

File: addons/spreadsheet/static/src/ir_ui_menu/index.js [hunk 2]
```diff
@@ -6,64 +6,141 @@ import spreadsheet from "@spreadsheet/o_spreadsheet/o_spreadsheet_extended";
...
+                            context: action.context,
+                        },
+                        { viewType }
+                    );
                 },
+                // createCell: (id, content, properties, sheetId, getters) => {
+                //     const { url } = parseMarkdownLink(content);
+                //     const actionDescription = parseViewLink(url);
+                //     return new OdooViewLinkCell(id, content, actionDescription, properties);
+                // },
```
Comment (pro-odoo) [inline addons/spreadsheet/static/src/ir_ui_menu/index.js:143]: Commented code

File: addons/spreadsheet/static/tests/links/dashboard_links_plugin_tests.js [hunk 3]
```diff
@@ -24,7 +25,7 @@ function beforeEach() {
     registry.category("services").add("orm", ormService, { force: true }); // #action-serv-leg-compat-js-class
 }
 
-QUnit.module("spreadsheet_dashboard > link", { beforeEach });
+QUnit.module("spreadsheet > link", { beforeEach });
```
Comment (pro-odoo) [inline addons/spreadsheet/static/tests/links/dashboard_links_plugin_tests.js:28]: Why ? We are still in spreadsheet_dashboard module

**Review by pro-odoo — COMMENTED** [review]
`[IMP] spreadsheet_*: update...`

File: addons/spreadsheet/static/tests/links/dashboard_links_plugin_tests.js (hunk 3 above)
Comment (LucasLefevre) [reply addons/spreadsheet/static/tests/links/dashboard_links_plugin_tests.js:28]: no.
The file name should be changed

**Pull request closed without merging**